### PR TITLE
mix format

### PR DESCRIPTION
**Pull request by SteffenDE**
There are still two files left that cannot easily be formatted currently.

One is the slot declaration in phoenix_component, where we don't want extra newlines, but there's not way to tell the formatter to omit them.

https://github.com/phoenixframework/phoenix_live_view/blob/f6e24c5bd7fc50fcb6dc74ac9d3df4c76bde7bd0/lib/phoenix_component.ex#L1017-L1023

The other is the html_engine_test file where we want to keep newlines that the formatter does not like.

Could we have a sigil_H modifier to prevent formatting? @josevalim

### CODE REVIEW

**Comment by josevalim** [comment]
> Could we have a sigil_H modifier to prevent formatting? @josevalim

We can definitely do it, but what to call it? "noformat"? A single "flag"? Will it be public or only internal?

File: lib/phoenix_live_view/html_formatter.ex [hunk 1]
```diff
@@ -225,30 +225,34 @@ defmodule Phoenix.LiveView.HTMLFormatter do
 
   @impl Mix.Tasks.Format
   def format(source, opts) do
-    line_length = opts[:heex_line_length] || opts[:line_length] || @default_line_length
-    newlines = :binary.matches(source, ["\r\n", "\n"])
-
-    formatted =
+    if opts[:sigil] === :H and opts[:modifiers] === ~c"noformat" do
```
Comment (josevalim) [inline lib/phoenix_live_view/html_formatter.ex:228]: Do we need to check the sigil? We register it specifically for ~H anyway, no?

Comment (josevalim) [reply lib/phoenix_live_view/html_formatter.ex:228]: Ignore me. We want to respect noformat only for ~H.

**Review by josevalim — APPROVED** [review]
Only one nit and ship it!

File: lib/phoenix_component.ex [hunk 2]
```diff
@@ -788,7 +788,7 @@ defmodule Phoenix.Component do
   for more information.
   '''
   @doc type: :macro
-  defmacro sigil_H({:<<>>, meta, [expr]}, []) do
+  defmacro sigil_H({:<<>>, meta, [expr]}, _modifiers) do
```
[the remark this answers is not in the document]
Comment (SteffenDE) [reply lib/phoenix_component.ex]: I added a guard. Now it will raise a FunctionClauseError as previously 👍

**Comment by SteffenDE** [comment]
🚀

**Pull request merged**